docs/design/2022-04-15-multi-schema-change.md

```diff
@@ -0,0 +1,137 @@
+# Proposal: Support Multi-Schema Change
+
+- Author(s): [tangenta](https://github.com/tangenta)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/14766
+
+## Abstract
+
+This proposes an implementation of Multi-Schema Change.
+
+## Background
+
+Multi-Schema Change is one of MySQL's extended features to the SQL standard. It allows the users to atomically make multiple schema changes in one SQL statement, including column and index `ADD`, `ALTER`, `DROP`, and `CHANGE`, as well as table option changes. For example:
+
+```sql
+CREATE TABLE t (a INT, c INT);
+ALTER TABLE t ADD COLUMN b INT AUTO_INCREMENT PRIMARY KEY, 
+ MODIFY COLUMN c CHAR(5),
+ ADD INDEX idx(a),
+ AUTO_INCREMENT = 1000;
+```
+
```

Comment:
The origins of this feature in MySQL explain why it is frequently used by applications:

- Historically, MySQL has performed DDL using a full copy of the table (although optimizations have since been added).
- Combining multiple DDL changes in one means that only one copy operation is required.

While this doesn't apply to TiDB's DDL, it does explain why it is pervasive syntax.